lib/es6-promise/promise/race.js

```diff
@@ -72,13 +72,13 @@ export default function race(entries) {
   let Constructor = this;
 
   if (!isArray(entries)) {
-    return new Constructor((_, reject) => reject(new TypeError('You must pass an array to race.')));
-  } else {
-    return new Constructor((resolve, reject) => {
-      let length = entries.length;
-      for (let i = 0; i < length; i++) {
-        Constructor.resolve(entries[i]).then(resolve, reject);
-      }
-    });
+    return Constructor.reject(new TypeError('You must pass an array to race.'));
```

Comment:
I don't believe the spec has this going through the public `Constructor.reject` method